### ISSUE TITLE
Parser incompatibility

### ISSUE THREAD

**Issue by jonas054** [bbatsov/rubocop#1806]
Version 2.2.0.4 introduced [some changes](https://github.com/whitequark/parse r/compare/v2.2.0.3...v2.2.0.4), one of which came from [this commit](https://github.com/whitequark/parser/commit/473e99008ad0f2fe9180bca9785e8e1b304da02d). It made a change in how comments are associated to code that affected the `Style/Documentation` cop. This is why [the build of RuboCop](https://travis-ci.org/bbatsov/rubocop) is currently failing.

The [affected code](https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/cop/style/documentation.rb#L90-L104) in `documentation.rb` is a workaround for a problem, I think, and the recent change in Parser is a correction. I'm unsure if the change follows the Semantic Versioning rules, but it makes no difference for us, given the way we specify the dependency:
```ruby
s.add_runtime_dependency('parser', '>= 2.2.0.1', '< 3.0')
```
We should just update `Style/Documentation` and change the dependency to
```ruby
s.add_runtime_dependency('parser', '>= 2.2.0.4', '< 3.0')
```

**Comment by bbatsov** [comment]
What about https://github.com/whitequark/parser/issues/190?

**Comment by bbatsov** [comment]
Or maybe I didn't read the other issue carefully enough. :-)

**Comment by jonas054** [comment]
Yeah, I think this one's in our own hands. I'll push a solution.
